RecoTracker/LSTCore/src/alpaka/Triplet.h

```diff
@@ -206,63 +206,304 @@ namespace lst {
+        py = -py;
+      if (charge == -1)
+        px = -px;
+    }
+    if (x_init < x_center && y_init > y_center)  //2nd quad
+    {
+      if (charge == -1) {
+        px = -px;
+        py = -py;
+      }
+    }
+    if (x_init < x_center && y_init < y_center)  //3rd quad
+    {
+      if (charge == 1)
+        px = -px;
+      if (charge == -1)
+        py = -py;
+    }
+    if (x_init > x_center && y_init < y_center)  //4th quad
+    {
+      if (charge == 1) {
+        px = -px;
+        py = -py;
+      }
+    }
+
+    //But if the initial T3 curve goes across quarters(i.e. cross axis to separate the quarters), need special redeclaration of px,py signs on these to avoid errors
+    if (x3 < x2 && x2 < x1)
+      px = -alpaka::math::abs(acc, px);
+    else if (x3 > x2 && x2 > x1)
+      px = alpaka::math::abs(acc, px);
+    if (y3 < y2 && y2 < y1)
+      py = -alpaka::math::abs(acc, py);
+    else if (y3 > y2 && y2 > y1)
+      py = alpaka::math::abs(acc, py);
+
+    float AO = alpaka::math::sqrt(acc, (x_other - x_center) * (x_other - x_center) + (y_other - y_center) * (y_other - y_center)); 
+    float BO = alpaka::math::sqrt(acc, (x_init - x_center) * (x_init - x_center) + (y_init - y_center) * (y_init - y_center));
+    float AB2 = (x_other - x_init) * (x_other - x_init) + (y_other - y_init) * (y_other - y_init); 
+    float dPhi = alpaka::math::acos(acc, (AO * AO + BO * BO - AB2) / (2 * AO * BO)); //Law of Cosines
+    float ds = circleRadius / 100 * dPhi;
+    float pz = dz / ds * pt; 
+
+    float p = alpaka::math::sqrt(acc, px * px + py * py + pz * pz);
+    float a = -2.f * k2Rinv1GeVf * 100 * charge;
+    float rou = a / p;
+
+    float rzChiSquared = 0;
+    float error = 0;
+
+    //check the tilted module, side: PosZ, NegZ, Center(for not tilted)
+    float drdz = alpaka::math::abs(acc, modulesInGPU.drdzs[outerOuterLowerModuleIndex]);
+    short side = modulesInGPU.sides[outerOuterLowerModuleIndex];
+    short subdets = modulesInGPU.subdets[outerOuterLowerModuleIndex];
+
+    //calculate residual
+    if (layer3 <= 6 && ((side == lst::Center) or (drdz < 1))) { // for barrel
+      float paraA = r_init * r_init + 2 * (px * px + py * py) / (a * a) + 2 * (y_init * px - x_init * py) / a - r3 * r3;
+      float paraB = 2 * (x_init * px + y_init * py) / a;
+      float paraC = 2 * (y_init * px - x_init * py) / a + 2 * (px * px + py * py) / (a * a);
+      float A = paraB * paraB + paraC * paraC;
+      float B = 2 * paraA * paraB;
+      float C = paraA * paraA - paraC * paraC;
+      float sol1 = (-B + alpaka::math::sqrt(acc, B * B - 4 * A * C)) / (2 * A);
+      float sol2 = (-B - alpaka::math::sqrt(acc, B * B - 4 * A * C)) / (2 * A);
+      float solz1 = alpaka::math::asin(acc, sol1) / rou * pz / p + z_init;
+      float solz2 = alpaka::math::asin(acc, sol2) / rou * pz / p + z_init;
+      float diffz1 = (solz1 - z3) * 100;
+      float diffz2 = (solz2 - z3) * 100;
+      if (alpaka::math::isnan(acc, diffz1))
```

Comment:
https://github.com/SegmentLinking/cmssw/pull/97

Please change the isnan calls in this PR to edm::isNotFinite.